src/fileop.c

```diff
@@ -56,20 +56,13 @@ static int librole_read_file_from_dir(const char const *directory,
 
     if (NULL == directory || NULL == filename || NULL == role_graph)
     {
-        goto librole_read_file_from_dir_end;
+        return retcode;
     }
 
     size_t dirlen = strlen(directory);
     size_t namelen = strlen(filename);
     size_t fullpathlen = dirlen + namelen + 1 + 1;
-    char *fullpath = NULL;
-
-    fullpath = calloc(fullpathlen, sizeof(char));
-    if (NULL == fullpath || 0 != errno)
-    {
-        retcode = errno;
-        goto librole_read_file_from_dir_end;
-    }
+    char fullpath[fullpathlen];
```

Comment:
This makes the code prone to stack overflow/smashing.
I think it's safer to allocate on heap (as the original version did).

Reply:
Ok, we could just check for max path length on stack.